CMakeLists.txt

```diff
@@ -66,6 +66,16 @@ if(BUILD_TOOLS)
         string(CONCAT RPATH "$ORIGIN/../" ${CMAKE_INSTALL_LIBDIR})
         set_target_properties(gltfpack PROPERTIES INSTALL_RPATH ${RPATH})
     endif()
+
+    if(EXISTS glfw)
```

Comment:
Is that the right way to discover glfw or is there some way to use find_package to do this? Not sure if this just checks for the folder.

Reply:
If we are building GLFW from the specific sources (which is the case for Windows), then this is pretty much in-line with the official documentation: https://www.glfw.org/docs/latest/build_guide.html (section `With CMake and GLFW source`).

The `find_package` approach is error-prone and unstable on Windows.

Reply:
I would need to look into this later; I'm not comfortable merging this as is because it feels very single-use and against the best practices of CMake that I know. The manual you link describes the find_package route as well, so I would ask if find_package can work on Windows given some external configuration.

If we are going down the source path route I'd want the path to be explicitly configurable and unset by default so that there's no assumption that glfw is in meshoptimizer's source tree.

Reply:
What if we check the user provided path first, then (if user provided no custom path) we do `find_package()`, and then we check the hardcoded `glfw` folder, and then, if everything else has failed, just skip building `lodviewer`? IMHO, this should cover Linux/Windows hosts pretty well.

Reply:
This seems reasonable. We could also start by simply changing the hardcoded glfw path to a (required) user input, and do the find_package dance separately.

I'm kinda hoping that long term CMake will evolve to make find_package route the only one necessary; it looks like today this might be feasible with vcpkg but it's not integrated into VS nicely so may not be super compelling just yet.